kubeflow/common/iap-ingress/base/config-map.yaml

```diff
@@ -181,6 +179,92 @@ data:
+    if [ -z ${PROJECT} ]; then
+      echo Error unable to fetch PROJECT from compute metadata
+      exit 1
+    fi
+
+    PROJECT_NUM=$(curl -s -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/project/numeric-project-id)
+    if [ -z ${PROJECT_NUM} ]; then
+      echo Error unable to fetch PROJECT_NUM from compute metadata
+      exit 1
+    fi
+
+    # Activate the service account
+    if [ ! -z "${GOOGLE_APPLICATION_CREDENTIALS}" ]; then
+      # As of 0.7.0 we should be using workload identity and never setting GOOGLE_APPLICATION_CREDENTIALS.
+      # But we kept this for backwards compatibility but can remove later.
+      gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS}
+    fi
+
+    # Print out the config for debugging
+    gcloud config list
+    gcloud auth list
+
+    set_endpoint () {
+      NODE_PORT=$(kubectl --namespace=${NAMESPACE} get svc ${SERVICE} -o jsonpath='{.spec.ports[?(@.name=="http2")].nodePort}')
+      echo "[DEBUG] node port is ${NODE_PORT}"
+
+      BACKEND_NAME=""
+      while [[ -z ${BACKEND_NAME} ]]; do
+        BACKENDS=$(kubectl --namespace=${NAMESPACE} get ingress ${INGRESS_NAME} -o jsonpath='{.metadata.annotations.ingress\.kubernetes\.io/backends}')
+        echo "[DEBUG] fetching backends info with ${INGRESS_NAME}: ${BACKENDS}"
+        BACKEND_NAME=$(echo $BACKENDS | grep -o "k8s-be-${NODE_PORT}--[0-9a-z]\+")
+        echo "[DEBUG] backend name is ${BACKEND_NAME}"
+        sleep 2
+      done
+
+      BACKEND_ID=""
+      while [[ -z ${BACKEND_ID} ]]; do
+        BACKEND_ID=$(gcloud compute --project=${PROJECT} backend-services list --filter=name~${BACKEND_NAME} --format='value(id)')
+        echo "[DEBUG] Waiting for backend id PROJECT=${PROJECT} NAMESPACE=${NAMESPACE} SERVICE=${SERVICE} filter=name~${BACKEND_NAME}"
+        sleep 2
+      done
+      echo BACKEND_ID=${BACKEND_ID}
+
+      JWT_AUDIENCE="/projects/${PROJECT_NUM}/global/backendServices/${BACKEND_ID}"
+      
+      # We use a regular expression to obtain the IP address of the target Ingress, assuming IPv4 standard.
+      INGRESS_TARGET_IP=$(kubectl get ingress --all-namespaces | grep -E -o "(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)")
+      
+      echo "[DEBUG] ENDPOINT_NAME = ${ENDPOINT_NAME}"
+      echo "[DEBUG] INGRESS_TARGET_IP = ${INGRESS_TARGET_IP}"
+      echo "[DEBUG] JWT_AUDIENCE = ${JWT_AUDIENCE}"
+      
+      # Create OpenAPI specification for the RESTful Cloud Endpoint
+      sed "s|JWT_AUDIENCE|${JWT_AUDIENCE}|;s|ENDPOINT_NAME|${ENDPOINT_NAME}|;s|INGRESS_TARGET_IP|${INGRESS_TARGET_IP}|" /var/envoy-config/swagger_template.yaml > openapi.yaml
+      
+      # Deploy and enable the endpoint
+      gcloud endpoints services deploy openapi.yaml
```

Comment:
In my setup, the admin SA does not have enough permissions to execute this.
I had to grant it `roles/serviceusage.serviceUsageAdmin` (see: https://cloud.google.com/service-usage/docs/access-control#predefined_roles)

Perhaps we need a new entry in `kf-admin-policy.yaml` ?

Reply:
After the solving the permission issue, I have a new deploy every 30 secs

![Screenshot from 2022-08-17 12-59-42](https://user-images.githubusercontent.com/308613/185010953-63d05c9e-0a86-4b94-89ab-f3390f7d9808.png)

Is this expected ?
Shouldn't we add a check and only deploy if necessary ?

Reply:
Hi @fabito.
Thank you for your feedback! 

The purpose of **cloud-endpoints-enabler** is to create and activate a cloud endpoint during deployment. It's supposed to be deleted at the end of `make apply` run. The behavior you observed is not intended, as we recommend running `make apply` and choose necessary components in **config.yaml** instead of deploying each component separately. 

As per permissions, I was not able to reproduce the error you mentioned. Could you create a separate issue with details about your GKE cluster?

Current approach clearly has room for improvement. Your contributions are very welcome!